nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -152,9 +243,9 @@ fi
 
 if [ -n "$buildNix" ]; then
     echo "building Nix..." >&2
-    if ! nix-build '<nixpkgs/nixos>' -A config.nix.package -o $tmpDir/nix "${extraBuildFlags[@]}" > /dev/null; then
-        if ! nix-build '<nixpkgs/nixos>' -A nixFallback -o $tmpDir/nix "${extraBuildFlags[@]}" > /dev/null; then
-            if ! nix-build '<nixpkgs>' -A nix -o $tmpDir/nix "${extraBuildFlags[@]}" > /dev/null; then
+    if ! nixBuild '<nixpkgs/nixos>' -A config.nix.package -o $tmpDir/nix "${extraBuildFlags[@]}" > /dev/null; then
```

Comment:
I am not sure, but I think I might prefer if we were to use `nix-instantiate` and `nix-store -r` explicitly here.
Thus this would just be a matter of prefixing all the commands with the location of the action.

```
  if ! drv=$(nix-instantiate …) ; then
    …
  fi
  if test -n "$drv"; then
    buildHostCmd nix-store -r … ;
  fi
```